frontend/app/services/subscription-service.server.ts

```diff
@@ -0,0 +1,70 @@
+import moize from 'moize';
+import { z } from 'zod';
+
+import { getAuditService } from '~/services/audit-service.server';
+import { getInstrumentationService } from '~/services/instrumentation-service.server';
+import { getEnv } from '~/utils/env.server';
+import { getLogger } from '~/utils/logging.server';
+
+const log = getLogger('subscription-service.server');
+
+/**
+ * Return a singleton instance (by means of memomization) of the letters service.
+ */
+export const getSubscriptionService = moize(createSubscriptionService, { onCacheAdd: () => log.info('Creating new letters service') });
+
+function createSubscriptionService() {
+  // prettier-ignore
+  const { 
+    INTEROP_API_BASE_URI,
+  } = getEnv();
+
+  async function getSubscription(sin: string, alertType: string) {
+    const auditService = getAuditService();
+    const instrumentationService = getInstrumentationService();
+    auditService.audit('alert-subscription.get', { sin });
+
+    const url = new URL(`${INTEROP_API_BASE_URI}/dental-care/applicant-information/dts/v1/subscriptions`);
+    url.searchParams.set('sin', sin);
+    url.searchParams.set('alertType', alertType);
+
+    const response = await fetch(url, {
+      // Using POST instead of GET due to how sin params gets logged with SIN
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+
+    instrumentationService.countHttpStatus('http.client.interop-api.alert-subscription.posts', response.status);
+
+    if (!response.ok) {
+      log.error('%j', {
+        message: 'Failed to fetch address',
+        status: response.status,
+        statusText: response.statusText,
+        url: url,
+        responseBody: await response.text(),
+      });
+
+      throw new Error(`Failed to fetch data. address: ${response.status}, Status Text: ${response.statusText}`);
+    }
+
+    if (response.status === 204) {
+      return null;
+    }
```

Comment:
We can remove this check as the API shouldn't return a 204.

Reply:
removed

Reply:
The service appears to be still handling the 204.

Reply:
Sorry, just realized this is for the `PUT` not the `GET`. Resolving.